src/main/java/csv/CsvReadProcessor.java

```diff
@@ -0,0 +1,53 @@
+package csv;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+
+public class CsvReadProcessor {
+
+    private String[] headers;
+    private String[] fields;
+    private String delimiter;
+    private BufferedReader csvBufferReader;
+    private FileReader csvFileReader;
+
+    public int openCsv(String inCsvFilePath,String initDelimiter) {
```

Comment:
```suggestion
    public List<String> readCsv(String inCsvFilePath) {
```
Давай в цьому методі зчитаємо всі строки у файлі і повернемо список строк у якості return type. Це погана практика зберігати інформацію в полях (private String[] headers, private String[] fields) і дописувати її з кожним новим викликом методу. І я не думаю, що нам треба передавати на вхід в метод роздільник, так як дані в csv файлах завжди розділені саме комою ( CSV: comma-separated values)